backend/repository/engine.repository.ts

```diff
@@ -0,0 +1,264 @@
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Update engine settings in the database.
+   * @param {string} id - The ID of the OIBus settings to update.
+   * @param {EngineSettingsCommandDTO} command - The command
+   */
+  updateEngineSettings(id: string, command: EngineSettingsCommandDTO): void {
+    const query =
+      `UPDATE ${ENGINE_TABLE} SET version = ?, name = ?, port = ?, ` +
+      `log_console_level = ?, ` +
+      `log_file_level = ?, ` +
+      `log_file_max_file_size = ?, ` +
+      `log_file_number_of_files = ?, ` +
+      `log_database_level = ?, ` +
+      `log_database_max_number_of_logs = ?, ` +
+      `log_loki_level = ?, ` +
+      `log_loki_interval = ?, ` +
+      `log_loki_address = ?, ` +
+      `log_loki_token_address = ?, ` +
+      `log_loki_proxy_id = ?, ` +
+      `log_loki_username = ?, ` +
+      `log_loki_password = ?, ` +
+      `health_signal_log_enabled = ?, ` +
+      `health_signal_log_interval = ?` +
+      `health_signal_http_enabled = ?, ` +
+      `health_signal_http_interval = ?, ` +
+      `health_signal_http_verbose = ?, ` +
+      `health_signal_http_address = ?, ` +
+      `health_signal_http_proxy_id = ?, ` +
+      `health_signal_http_authentication_type = ?, ` +
+      `health_signal_http_authentication_key = ?, ` +
+      `health_signal_http_authentication_secret = ? ` +
+      `WHERE id = ?`;
+    this.database
+      .prepare(query)
+      .run(
+        command.version,
+        command.name,
+        command.port,
+        command.logParameters.console.level,
+        command.logParameters.file.level,
+        command.logParameters.file.maxFileSize,
+        command.logParameters.file.numberOfFiles,
+        command.logParameters.database.level,
+        command.logParameters.database.maxNumberOfLogs,
+        command.logParameters.loki.level,
+        command.logParameters.loki.interval,
+        command.logParameters.loki.address,
+        command.logParameters.loki.tokenAddress,
+        command.logParameters.loki.proxyId,
+        command.logParameters.loki.username,
+        command.logParameters.loki.password,
+        command.healthSignal.logging.enabled,
+        command.healthSignal.logging.interval,
+        command.healthSignal.http.enabled,
+        command.healthSignal.http.interval,
+        command.healthSignal.http.verbose,
+        command.healthSignal.http.address,
+        command.healthSignal.http.proxyId,
+        command.healthSignal.http.authentication.type,
+        command.healthSignal.http.authentication.key,
+        command.healthSignal.http.authentication.secret,
+        id
+      );
+  }
+
+  /**
+   * Create engine settings in the database.
+   * @param {string} id - The ID of the OIBus settings to create. It is not auto generated since it can be shared with other OIBus
+   * @param {EngineSettingsCommandDTO} command - The command
+   */
+  createEngineSettings(id: string, command: EngineSettingsCommandDTO): void {
+    const query =
```

Comment:
we might need to check if a line is already there for the given ID and throw an exception or delete the line or use upsert?